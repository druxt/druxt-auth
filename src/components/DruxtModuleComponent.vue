<script>
import DruxtModule from 'druxt/dist/components/DruxtModule.vue'

export default {
  // @TODO - Rename your module's component name.
  name: 'DruxtModuleComponent',

  extends: DruxtModule,

  data: () => ({
    foo: undefined,
    settings: undefined,
  }),

  /**
   * Druxt hooks.
   */
  druxt: {
    /**
     * Load configuration data.
     */
    async fetchConfig() {
      this.foo = 'bar'
    },

    /**
     * Load data using the provided settings.
     *
     * @param {object} settings - The settings object for this Druxt module.
     */
    async fetchData(settings) {
      this.settings = settings
    },

    /**
     * Create component options for Druxt Wrapper component.
     *
     * @param {object} context - The Druxt module context object.
     */
    componentOptions: (context) => ([
      [context.foo, 'default'],
      ['default']
    ]),

    /**
     * Create an object of props data to be used by the Druxt Wrapper component.
     *
     * @param {object} context - The Druxt module context object.
     */
    propsData: (context) => ({
      foo: context.foo
    }),

    /**
     * Create an object of scoped slots to be used by the Druxt Wrapper component.
     *
     * @param {function} h - The Vue hyperscript function.
     */
    slots(h) {
      return {
        default: () => h('div', ['Hello world'])
      }
    }
  }
}
</script>
