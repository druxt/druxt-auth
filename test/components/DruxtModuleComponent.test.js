import 'regenerator-runtime/runtime'
import { createLocalVue, mount } from '@vue/test-utils'

import DruxtWrapper from 'druxt/dist/components/DruxtWrapper.vue'
import DruxtModuleComponent from '../../src/components/DruxtModuleComponent.vue'

// Setup local vue instance.
const localVue = createLocalVue()
localVue.component('DruxtWrapper', DruxtWrapper)

// Mount the Vue component.
const mountComponent = function() {
  return mount(
    DruxtModuleComponent,
    {
      localVue,
      mocks: {
        $fetchState: { pending: true }
      }
    }
  )
}

describe('DruxtModuleComponent', () => {
  test('Hello world', async () => {
    // Mount the component.
    const wrapper = mountComponent()

    // Simulate the Nuxt fetch hook.
    await wrapper.vm.$options.fetch.call(wrapper.vm)

    // Assert data and props are as expected.
    expect(wrapper.vm.foo).toBe('bar')
    expect(wrapper.vm.component).toMatchSnapshot()

    // Assert slots.
    const h = jest.fn()
    const slots = wrapper.vm.$options.druxt.slots(h)
    slots.default()
    expect(h).toBeCalledWith('div', ['Hello world'])
  })
})
