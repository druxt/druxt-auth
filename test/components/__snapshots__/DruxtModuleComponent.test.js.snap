// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DruxtModuleComponent Hello world 1`] = `
Object {
  "$attrs": Object {
    "foo": "bar",
  },
  "is": "DruxtWrapper",
  "options": Array [
    "DruxtModuleComponentBarDefault",
    "DruxtModuleComponentBar",
    "DruxtModuleComponentDefault",
  ],
  "props": Object {},
  "propsData": Object {
    "foo": "bar",
  },
  "settings": Object {},
  "slots": Array [
    "default",
  ],
}
`;
